tests/test_subscriptionmanager.py

```diff
@@ -106,6 +106,21 @@ def test_hypervisorCheckInAsync(self, rhsmconnection):
                         "Only one will be recorded at the server.")
         self.sm.connection.return_value.has_capability = MagicMock(return_value=False)
 
+    @patch('rhsm.connection.UEPConnection')
+    # def test_hypervisorHeartbeat(self):
+    def test_hypervisorHeartbeat(self, rhsmconnection):
+        owner = 'owner'
+        config = VirtConfigSection.from_dict({'type': 'libvirt', 'owner': owner}, 'test', None)
+        # Ensure we try out the new API
+        rhsmconnection.return_value.has_capability.return_value = True
+        self.sm.logger = MagicMock()
+        self.sm.hypervisorHeartbeat(config, options=None)
+        self.sm.connection.hypervisorHeartbeat.assert_called_with(
+            owner,
+            None
+        )
+        self.sm.connection.return_value.has_capability = MagicMock(return_value=False)
+
```

Comment:
I would add here two more unit tests.
* First unit test for covering the case, when candlepin server does not have the capability.
* Second unit test for the case there are some options: `self.sm.hypervisorHeartbeat(config, options={'global':{'foo': 'bar'}})`

Reply:
this test must be done in Subscription Manager. The connection object is mocked and I cannot test the condition in the connection method from virt-who. Will add subman tests.

Reply:
Good point. I just reviewed and merged the PR: https://github.com/candlepin/subscription-manager/pull/2101